src/analyzer/schema-generator.ts

```diff
@@ -179,6 +179,14 @@ async function generatePageModelsForFiles({
 }: GeneratePageModelsOptions): Promise<{ pageModels: PartialPageModelWithFilePaths[]; objectModels: PartialObjectModel[] }> {
     let pageModels: PartialPageModel[] = [];
     let modelNameCounter = 1;
+
+    if (filePaths.length === 0) {
+        return {
+            pageModels: [],
+            objectModels: []
+        }
+    }
+
```

Comment:
the problem is not here, it down the lines after we generate `pageModels` and map them to flattened `filePaths`.